ggml/src/ggml-cuda/dmmv.cu

```diff
@@ -476,13 +477,31 @@ static __global__ void dequantize_mul_mat_vec(const void * __restrict__ vx, cons
             // matrix multiplication
             // for qr = 2 the y index needs to increase by 1 per j iter because of y_offset = qk/2
 #ifdef GGML_CUDA_F16
-            tmp += __hmul2(v, {
-                y[iybs + iqs + j/qr + 0],
-                y[iybs + iqs + j/qr + y_offset]
-            });
+            if ( y_offset == 1 ) {
+                // load 2 dfloats into register in a single instruction
+                const dfloat2 y_reg = *((dfloat2 *) &(y[iybs + iqs + j/qr]));
+                tmp += __hmul2(v, {
+                        y_reg.x;
+                        y_reg.y;
+                    });
```

Comment:
```suggestion
                tmp += __hmul2(v, y_reg);
```

If `GGML_CUDA_F16` is defined then `dfloat2` is `half2` so this should compile. Without this change the code will also again not work with HIP because the `.x` and `.y` operators on `half2` don't work properly.